### ISSUE TITLE
[Rule Request] Prefer `#unavailable`

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#3897]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

[SE-0290](https://github.com/apple/swift-evolution/blob/main/proposals/0290-negative-availability.md) introduces `#unavailable`, a new way to handle negative availability check. This can be useful when we're only interested in the `else` of an availability check.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.3. 

```swift
// trigger
if #available(iOS 14.0) {

} else {
  oldIos13TrackingLogic(isEnabled: ASIdentifierManager.shared().isAdvertisingTrackingEnabled)
}

if #available(iOS 14.0) {
  // we don't need to do anything here
} else {
  oldIos13TrackingLogic(isEnabled: ASIdentifierManager.shared().isAdvertisingTrackingEnabled)
}

if #available(iOS 13, *) {} else {
  loadMainWindow()
}


// ok
if #unavailable(iOS 13) {
  loadMainWindow()
}

if #available(iOS 9.0, *) {
  doSomething()
} else {
  legacyImplementation()
}
```

4. Should the rule be configurable, if so what parameters should be configurable?5. 

Just severity.
6. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Probably enabled by default on Swift 5.6+. I imagine that there won't be that many violations of this rule in existing codebases to make this disruptive.
